internal/extender/resourcereservations.go

```diff
@@ -283,6 +386,27 @@ func (rrm *ResourceReservationManager) getActivePods(ctx context.Context, pod *v
 	return activePodNames, nil
 }
 
+func (rrm *ResourceReservationManager) onExecutorPodDeletion(obj interface{}) {
+	pod, ok := utils.GetPodFromObjectOrTombstone(obj)
+	if !ok {
+		rrm.logger.Warn("failed to parse object as pod, skipping")
```

Comment:
lets keep the only usage of `rrm.logger` here because we don't have a context, but for the rest of the logs, lets use the `ctx` we get from the predicate call

Reply:
Discussed and decided to keep as is since it might be less confusing to have the compaction logs be separate from the predicate context and the params added to it. My view is that the compaction is a 'background' process, and the only reason to currently initiate it from the Predicate call is that we cannot be sure we are the leader until we get a request.